pubsub/google/cloud/pubsub/_http.py

```diff
@@ -203,9 +204,10 @@ def topic_publish(self, topic_path, messages):
         :rtype: list of string
         :returns: list of opaque IDs for published messages.
         """
-        _transform_messages_base64(messages, _base64_unicode)
+        messages_to_send = copy.deepcopy(messages)
```

Comment:
Got it, so the caller sends a shallow copy: `self.messages[:]`.

This fix is just fine, though I wonder if we should just create a new dictionary on the fly in `_transform_messages_base64`. It's already doing a loop over all the data, so the copy effectively makes us do that loop twice.